btcjson/chainsvrcmds.go

```diff
@@ -163,6 +163,24 @@ func NewGetBlockCountCmd() *GetBlockCountCmd {
 	return &GetBlockCountCmd{}
 }
 
+// GetBlockFilterCmd defines the getblockfilter JSON-RPC command.
+type GetBlockFilterCmd struct {
+	BlockHash  string  // The hash of the block
```

Comment:
I'd suggest using the `chainhash.Hash` type here.

Reply:
These `btcjson.*Cmd` structs are generally using basic types assigned through NewCmd constructors. `btcjson.assignField` function makes a conversion from command-line input to the basic type that NewCmd takes.

While it's possible to implement JSON Unmarshaler to use a more specific type as you suggest I think the idea is to keep these types as stupid as possible.

In the JSON-RPC request the type will be string that the server has to unmarshal. Just taking the string from command-line and converting it to a JSON-RPC request without any semantic validation offloads the validation to server. Doing validation client-side would result in double validation which would be more fragile and more complex to consume.

Reply:
`rpcclient.GetBlockFilter` now takes `chainhash.Hash`. `rpcclient` is where to take the Golang call and transform it to JSON.